components/brave_vpn/browser/brave_vpn_metrics.cc

```diff
@@ -112,4 +128,39 @@ void BraveVpnMetrics::HandleShowWidgetChange() {
   UMA_HISTOGRAM_BOOLEAN(kHideWidgetHistogramName, true);
 }
 
+void BraveVpnMetrics::RecordVPNConnectedInterval() {
+  connected_minutes_storage_.AddDelta(kConnectionReportInterval.InMinutes());
+}
+
+void BraveVpnMetrics::ReportVPNConnectedDuration() {
+  connection_report_timer_.Start(
+      FROM_HERE, base::Time::Now() + kConnectionReportInterval,
+      base::BindOnce(&BraveVpnMetrics::ReportVPNConnectedDuration,
+                     base::Unretained(this)));
+
+  if (!delegate_->is_purchased_user() || !uptime_monitor_) {
+    return;
+  }
+
+#if !BUILDFLAG(IS_ANDROID)
+  if (delegate_->IsConnected() && uptime_monitor_->IsInUse()) {
+    RecordVPNConnectedInterval();
```

Comment:
`BraveVpnService` is per-profile service. so browser could have multiple instances.
Maybe, most (all) of users uses one profile so we could ignore it.
but curious is it ok to run this record function from multiple service instance?

Reply:
according to stats, 99% of users only use one profile, so users with multiple VPN-subscribed profiles should not skew the stats significantly